src/schema-parser/asyncapi-schema-parser.ts

```diff
@@ -8,12 +19,73 @@ export function AsyncAPISchemaParser(): SchemaParser {
   }
 }
 
-function validate() {
+async function validate(input: ValidateSchemaInput<unknown, unknown>): Promise<SchemaValidateResult[]> {
+  const version = input.asyncapi.semver.version
+  const validator = findSchemaValidator(version);
+
+  let result: SchemaValidateResult[] = []
+  const valid = validator(input.data);
+  if (!valid && validator.errors) {
+    result = ajvToSpectralResult(validator.errors, input.path);
```

Comment:
That function will run in async mode, and `validator` saves errors to this same validator object (between execution), so better option will be to copy the errors to new array, for safety.

```suggestion
    result = ajvToSpectralResult([...validator.errors], input.path);
```

Reply:
nice catch! fixing